### PR TITLE
fix: core data add gradual migration [WPB-5591]

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5591" title="WPB-5591" target="_blank">BugWPB-5591</a>  [iOS] Not enough disk space in app
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

This is a cherry-pick from release branch release/cycle-3.111

### Notes (Optional)

* Manual resolution of conflicts
* Remove deprecated iOS 15 methods

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]

File: wire-ios-sync-engine/Source/SessionManager/SessionManager.swift [hunk 1]
```diff
@@ -244,6 +244,8 @@ public final class SessionManager: NSObject, SessionManagerType {
     let configuration: SessionManagerConfiguration
     var pendingURLAction: URLAction?
     let apiMigrationManager: APIMigrationManager
+    var cryptoboxMigrationManager: CryptoboxMigrationManagerInterface = CryptoboxMigrationManager()
```
Comment (markusfassbender) [inline wire-ios-sync-engine/Source/SessionManager/SessionManager.swift]: Seems like `cryptoboxMigrationManager` is not used in this file. Is this intended?

File: wire-ios-data-model/Source/ManagedObjectContext/Migration/CoreDataMessagingMigrationVersion.swift [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    private enum Constant {
+        static let dataModelPrefix = "zmessaging"
+        static let modelDirectory = "zmessaging.momd"
+        static let resourceExtension = "mom"
+    }
+
+    // MARK: -
+
+    // Note: add new versions here in first position!
+    case version2_109 = "zmessaging2.109.0"
```
Comment (markusfassbender) [inline wire-ios-data-model/Source/ManagedObjectContext/Migration/CoreDataMessagingMigrationVersion.swift:32]: I think in the original merge in `release` there was a new `2_210` version added. Was it dropped or will be added later in another PR (don't have the overview right now).

**Review by markusfassbender — APPROVED** [review]

File: wire-ios-data-model/Source/ManagedObjectContext/Migration/CoreDataMessagingMigrationVersion.swift (hunk 2 above)
Comment (netbe) [reply wire-ios-data-model/Source/ManagedObjectContext/Migration/CoreDataMessagingMigrationVersion.swift:32]: added in next PR

File: wire-ios-sync-engine/Source/SessionManager/SessionManager.swift (hunk 1 above)
Comment (netbe) [reply wire-ios-sync-engine/Source/SessionManager/SessionManager.swift]: nice catch

**Pull request merged**